tasks/discover-granules/tests/index.js

```diff
@@ -116,60 +115,150 @@ test('discover granules using HTTP', async (t) => {
+    }
+  };
 
 test('discover granules using S3',
-  discoverGranulesUsingS3((config) => {
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    config.provider = {
+      id: 'MODAPS',
+      protocol: 's3',
+      host: config.sourceBucketName
+    };
+  }));
+
+test('discover granules without collection files config using S3',
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    // Without files config we should still discover granules, but the
+    // discovered granules will have empty files arrays.
+    config.collection.files = [];
+    config.provider = {
+      id: 'MODAPS',
+      protocol: 's3',
+      host: config.sourceBucketName
+    };
+  }, (t, output) => {
+    validateOutput(t, output);
+    t.is(output.granules.length, 3);
+    output.granules.forEach(({ files }) => t.is(files.length, 0));
+  }));
+
+test('discover granules without collection files config, but configuring collection to ignore it, using S3',
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    // Without files config we should still discover granules, and the
+    // discovered granules' files arrays will include all files because we're
+    // ignoring the (empty) files config for filtering files.
+    config.collection.files = [];
+    config.collection.ignoreFilesConfig = true;
+    config.provider = {
+      id: 'MODAPS',
+      protocol: 's3',
+      host: config.sourceBucketName
+    };
+  }, (t, output) => {
+    validateOutput(t, output);
+    t.is(output.granules.length, 3);
+    output.granules.forEach(({ files }) => t.is(files.length, 2));
+  }));
+
+test('discover granules without collection files config, but configuring task to ignore it, using S3',
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    // Without file configs we should still discover granules, and the
+    // discovered granules files arrays will include all files because we're
+    // ignoring the (empty) files config for filtering files.
+    config.collection.files = [];
+    config.ignoreFilesConfig = true;
+    config.provider = {
+      id: 'MODAPS',
+      protocol: 's3',
+      host: config.sourceBucketName
+    };
+  }, (t, output) => {
+    validateOutput(t, output);
+    t.is(output.granules.length, 3);
+    output.granules.forEach(({ files }) => t.is(files.length, 2));
+  }));
+
+test('discover granules without collection files config, but configuring task to ignore it and overriding collection config not to ignore it, using S3',
+  discoverGranulesUsingS3(({ context: { event: { config } } }) => {
+    // Without files config we should still discover granules, and the
+    // discovered granules files arrays should be empty even though the task is
+    // configured to ignore the files config, but collection setting overrides
+    // it to NOT ignore the files config.
+    config.collection.files = [];
+    config.ignoreFilesConfig = true;
```

Comment:
I feel the opposite.  If the task configuration `ignoreFilesConfig = true`, it overwrites the configuration from `collection`.

Reply:
You have to look at lines 146-147 in `packages/ingest/granule.js` to see that the collection setting overrides the task setting, which is the desired behavior. The task setting behaves as the ("global") default across all collections, except where a collection explicitly sets a value for the property in order to override the "global" default.

Reply:
I've updated the PR to reverse the override so that the task setting overrides the collection setting.